android-json-form-wizard/src/main/java/com/vijay/jsonwizard/utils/Utils.java

```diff
@@ -874,6 +882,38 @@ public static int getResourceId(Context context, String name, ResourceType resou
     public static boolean isEmptyJsonArray(JSONArray jsonArray) {
         return jsonArray == null || jsonArray.length() == 0;
     }
+
+    /**
+     * Returns the object that holds the repeating group count
+     *
+     * @return
+     * @throws JSONException
+     */
+    @Nullable
+    public static JSONObject getRepeatingGroupCountObj(@NotNull WidgetArgs widgetArgs) throws JSONException {
+        String repeatingGroupCountObjKey = widgetArgs.getJsonObject().get(KEY) + "_count";
+        JSONObject stepJsonObject = widgetArgs.getFormFragment().getStep(widgetArgs.getStepName());
+        if (stepJsonObject != null) {
+            JSONArray stepFields = stepJsonObject.optJSONArray(JsonFormConstants.FIELDS);
+            JSONObject repeatingGroupCountObj = FormUtils.getFieldJSONObject(stepFields, repeatingGroupCountObjKey);
+            // prevents re-adding the count object during form traversals
+            if (repeatingGroupCountObj != null) {
+                return repeatingGroupCountObj;
+            }
+
+            repeatingGroupCountObj = new JSONObject();
+            repeatingGroupCountObj.put(KEY, repeatingGroupCountObjKey);
+            repeatingGroupCountObj.put(OPENMRS_ENTITY_PARENT, "");
+            repeatingGroupCountObj.put(OPENMRS_ENTITY, "");
+            repeatingGroupCountObj.put(OPENMRS_ENTITY_ID, "");
+            repeatingGroupCountObj.put(TYPE, "");
+            repeatingGroupCountObj.put(TEXT, widgetArgs.getJsonObject().get(REFERENCE_EDIT_TEXT_HINT));
+            stepFields.put(repeatingGroupCountObj);
+            return repeatingGroupCountObj;
+        } else {
+            return null;
```

Comment:
Do we need the else block? Is there ever an instance when `stepJsonObject` is null?

Reply:
yeah for malformed json form

Reply:
The method throws JSONException, so I'm thinking malformed JSON will be covered here? The calling method can then figure out how to handle the exception, no?

Reply:
That is a NPE check.

Reply:
Ok, but when is the `stepJsonObject` null?

Reply:
either way, you can simply return null without the else block, right?

Reply:
`Ok, but when is the stepJsonObject null?`
When the step does not exist ... for malformed json forms

`either way, you can simply return null without the else block, right?` 
I will still need an if block

Reply:
yea, we can remove the else block though.